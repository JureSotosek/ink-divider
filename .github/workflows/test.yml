name: ğŸ§ª Tests

on:
  # https://blog.esciencecenter.nl/reduce-reuse-recycle-save-the-planet-one-github-action-at-a-time-4ab602255c3f
  push:
    paths:
      - src/*

  pull_request:
    paths:
      - src/*

  workflow_dispatch:

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: ğŸ“¥ Checkout
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2

      - name: ğŸ”§ Setup Bun
        uses: oven-sh/setup-bun@4bc047ad259df6fc24a6c9b0f9a0cb08cf17fbe5 # v2
        with:
          bun-version-file: ".bun-version"

      - name: ğŸ“¦ Install dependencies
        run: bun install --frozen-lockfile

      - name: ğŸ§ª Run tests
        run: bun test
