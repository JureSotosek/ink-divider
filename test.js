const { h, Color, renderToString } = require('ink');
const test = require('ava');

const Divider = require('.');

test('Basic divider', t => {
  const myDivider = renderToString(<Divider title="Title" />);

  const expected = renderToString(
    <div>
      {' '}
      <Color keyword={'grey'}>{'─────────────────────'}</Color>
      <Color keyword={'white'}>{' Title '}</Color>
      <Color keyword={'grey'}>{'─────────────────────'}</Color>{' '}
    </div>
  );

  t.is(myDivider, expected);
});
