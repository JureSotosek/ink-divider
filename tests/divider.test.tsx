import { expect, test } from 'bun:test'

import { Box, Text } from 'ink'
import { render } from 'ink-testing-library'

import Divider from '../src'

test('divider in box with sample text', () => {
	const lorem = (
		<Text>Lorem ipsum dolor sit amet consectetur adipisicing elit</Text>
	)
	const { lastFrame } = render(
		<Box
			borderStyle="round"
			flexDirection="column"
			alignSelf="flex-start"
			paddingLeft={1}
			paddingRight={1}
		>
			{lorem}
			<Divider title="Title" />
			{lorem}
		</Box>
	)

	expect(lastFrame()).toMatchSnapshot()
})

test('divider without title', () => {
	const { lastFrame } = render(<Divider />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with title', () => {
	const { lastFrame } = render(<Divider title="Title" />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with custom width', () => {
	const { lastFrame } = render(<Divider width={20} />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with custom padding', () => {
	const { lastFrame } = render(<Divider padding={2} />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with custom title and width', () => {
	const { lastFrame } = render(<Divider title="Title" width={20} />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with title and custom titlePadding', () => {
	const { lastFrame } = render(<Divider title="Title" titlePadding={3} />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with title and custom titleColor', () => {
	const { lastFrame } = render(<Divider title="Title" titleColor="blue" />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with custom dividerChar', () => {
	const { lastFrame } = render(<Divider dividerChar="=" />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with custom dividerColor', () => {
	const { lastFrame } = render(<Divider dividerColor="red" />)
	expect(lastFrame()).toMatchSnapshot()
})

test('divider with all custom props', () => {
	const { lastFrame } = render(
		<Divider
			title="Custom Title"
			width={30}
			padding={1}
			titlePadding={2}
			titleColor="green"
			dividerChar="*"
			dividerColor="yellow"
		/>
	)
	expect(lastFrame()).toMatchSnapshot()
})
