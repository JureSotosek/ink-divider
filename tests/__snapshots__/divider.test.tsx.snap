// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`divider in box with sample text 1`] = `
"╭─────────────────────────────────────────────────────────╮
│ Lorem ipsum dolor sit amet consectetur adipisicing elit │
│ \x1B[90m────────────────────────\x1B[39m \x1B[37mTitle\x1B[39m \x1B[90m────────────────────────\x1B[39m │
│ Lorem ipsum dolor sit amet consectetur adipisicing elit │
╰─────────────────────────────────────────────────────────╯"
`;

exports[`divider without title 1`] = `"\x1B[90m────────────────────────────────────────────────────────────────────────────────────────────────────\x1B[39m"`;

exports[`divider with title 1`] = `"\x1B[90m───────────────────────────────────────────────\x1B[39m \x1B[37mTitle\x1B[39m \x1B[90m──────────────────────────────────────────────\x1B[39m"`;

exports[`divider with custom width 1`] = `"\x1B[90m────────────────────────────────────────────────────────────────────────────────────────────────────\x1B[39m"`;

exports[`divider with custom padding 1`] = `"  \x1B[90m────────────────────────────────────────────────────────────────────────────────────────────────\x1B[39m"`;

exports[`divider with title and custom titlePadding 1`] = `"\x1B[90m─────────────────────────────────────────────\x1B[39m   \x1B[37mTitle\x1B[39m   \x1B[90m────────────────────────────────────────────\x1B[39m"`;

exports[`divider with title and custom titleColor 1`] = `"\x1B[90m───────────────────────────────────────────────\x1B[39m \x1B[34mTitle\x1B[39m \x1B[90m──────────────────────────────────────────────\x1B[39m"`;

exports[`divider with custom dividerChar 1`] = `"\x1B[90m====================================================================================================\x1B[39m"`;

exports[`divider with custom dividerColor 1`] = `"\x1B[31m────────────────────────────────────────────────────────────────────────────────────────────────────\x1B[39m"`;

exports[`divider with all custom props 1`] = `" \x1B[33m******\x1B[39m  \x1B[32mCustom Title\x1B[39m  \x1B[33m******\x1B[39m"`;

exports[`divider with title and other box props (flexGrow, etc.) 1`] = `" \x1B[37mTitle\x1B[39m"`;
